main.go

```diff
@@ -563,18 +563,15 @@ func doReload() {
 	log.WithFields(logrus.Fields{
 		"prefix": "main",
 	}).Info("Preparing new router")
-	newRouter := mux.NewRouter()
-	mainRouter = newRouter
+	mainRouter = mux.NewRouter()
+	if globalConf.HttpServerOptions.OverrideDefaults {
```

Comment:
This is also a new addition, which I proposed in https://github.com/TykTechnologies/tyk/pull/1094.
Since we creating a new router, we should re-apply same router settings.

Reply:
I wonder how this ever worked properly before this change.

Reply:
It did not worked :)